### PR TITLE
Upgrade to PTL 1.6.0

### PR DESCRIPTION
**Pull request by ericharper**
# What does this PR do ?

This PR will have fixes needed for PTL 1.6.0. The PR will first install PTL 1.6.0rc0 via the Jenkinsfile. Then when PTL 1.6.0 is on PYPI, PTL will be installed via .reinstall.sh. Do not merge this PR until PTL 1.6.0 is on PYPI.

**Collection**: Core

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/nlp/language_modeling/megatron_bart_pretraining.py [hunk 1]
```diff
@@ -64,10 +64,10 @@ def main(cfg) -> None:
     exp_manager(trainer, cfg.exp_manager)
 
     # update resume from checkpoint found by exp_manager
-    resume_from_checkpoint = trainer.checkpoint_connector.resume_from_checkpoint_fit_path
+    resume_from_checkpoint = trainer._checkpoint_connector.resume_from_checkpoint_fit_path
```
Comment (titu1994) [inline examples/nlp/language_modeling/megatron_bart_pretraining.py:67]: Todo: Ask PTL to provide a stable public API for this

File: nemo/core/classes/modelPT.py [hunk 2]
```diff
@@ -1030,7 +1030,7 @@ def maybe_init_from_pretrained_checkpoint(self, cfg: OmegaConf, map_location: st
                         trainer = self.trainer
                         if (
                             hasattr(trainer, 'resume_from_checkpoint')
-                            and trainer.checkpoint_connector.resume_checkpoint_path is not None
+                            and trainer._checkpoint_connector.resume_checkpoint_path is not None
```
Comment (titu1994) [inline nemo/core/classes/modelPT.py:1033]: Todo: ask PTL team for stable API for changing connectors

File: requirements/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
+setuptools==59.5.0
```
Comment (titu1994) [inline requirements/requirements.txt:1]: Todo: revert this when setuptools fixes itself.

**Review by titu1994 — APPROVED** [review]
Overall looks good, thanks ! Ive added a few todos we need to go back to - we need to revert setuptools patch next month (maybe) and definitely need to ask PTL for a stable way of accessing and overriding the connectors. 

Accessing private variables inside ModelPT and expmanager is simply asking for bugs in the future

File: examples/nlp/language_modeling/megatron_bert_pretraining.py [hunk 4]
```diff
@@ -47,10 +47,11 @@ def main(cfg) -> None:
     exp_manager(trainer, cfg.exp_manager)
 
     # update resume from checkpoint found by exp_manager
-    resume_from_checkpoint = trainer.checkpoint_connector.resume_from_checkpoint_fit_path
+    resume_from_checkpoint = trainer._checkpoint_connector.resume_from_checkpoint_fit_path
+    # resume_from_checkpoint = uninject_model_parallel_rank(resume_from_checkpoint)
```
Comment (MaximumEntropy) [inline examples/nlp/language_modeling/megatron_bert_pretraining.py:51]: We could uncomment this?

File: tests/core/test_exp_manager.py [hunk 5]
```diff
@@ -185,7 +185,7 @@ def test_trainer_loggers(self, tmp_path):
                 "create_checkpoint_callback": False,
                 "exp_dir": str(tmp_path),
                 "create_wandb_logger": True,
-                "wandb_logger_kwargs": {"name": "", "project": ""},
+                "wandb_logger_kwargs": {"name": "", "project": "", "offline": True},
```
Comment (MaximumEntropy) [inline tests/core/test_exp_manager.py:188]: Does this mean every exp_manager config needs to set this to True to log stuff to your wandb account?

**Review by MaximumEntropy — COMMENTED** [review]
LGTM! Just a couple of comments/questions.

**Review by MaximumEntropy — APPROVED** [review]
LGTM. Misunderstood the wandb part.

**Pull request merged**